cmd/kubectl.go

```diff
@@ -74,3 +117,75 @@ func runKubectl(ctx *cli.Context) error {
 	}
 	return nil
 }
+
+func loadKubeconfig(path string) ([]byte, error) {
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return content, err
+}
+
+func writeKubeconfig(filePath string, kubeconfig []byte) error {
+	err := os.MkdirAll(path.Dir(filePath), 0700)
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Saving config to %s", filePath)
+
+	if err := os.Remove(filePath); err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+	}
+
+	output, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer output.Close()
+
+	_, err = output.Write(kubeconfig)
+	return err
+}
+
+func extractKubeconfigTokenID(kubeconfig []byte) (string, error) {
+	type Token struct {
+		Users []struct {
+			User struct {
+				T string `yaml:"token"`
+			} `yaml:"user"`
+		} `json:"users,omitempty" yaml:"users,omitempty"`
+	}
```

Comment:
Should this be declared outside of the function?

Reply:
I considered this as well, I decided to contain it within the function because it's strictly being used to unmarshal, and only to extract the token at that so idk if it would ever have a user outside of the function since it is incomplete.

Reply:
I see. Is there any way we could use the structs defined in [`client-go/tools/clientcmd`](https://github.com/kubernetes/client-go/blob/30548acd0a9e231d42d406433aef55b111b36240/tools/clientcmd/api/v1/types.go#L28-L52) instead?

We could use [LoadFromFile](https://godoc.org/k8s.io/client-go/tools/clientcmd#LoadFromFile), [IsEmptyConfig](https://godoc.org/k8s.io/client-go/tools/clientcmd#IsEmptyConfig), and [WriteToFile](https://godoc.org/k8s.io/client-go/tools/clientcmd#WriteToFile). This might allow us to eliminate / simplify `loadKubeConfig`, `writeKubeConfig`, and `extractKubeconfigTokenID` here without requiring an additional inline struct.

Reply:
sure, I can do- will need to add client-go as a dependency though.